dev/diagrams/README.md

```diff
@@ -0,0 +1,26 @@
+<!--
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+ -->
+
+In order to regenerate the diagrams you need to:
+
+* Have [Graphviz](https://graphviz.org/) installed on your system
+* Create and activate a virtual environment of yours
```

Comment:
Actually  I thought a bit on that and I will move it to pre-commit, i need to add a bit more checks there (not sure if the generation is reproducible so I will likely need to add some hash calculation - but it will be better because it will not require to setup the venve at all (will be handled by pre-commit automatically)